core/common/src/main/java/alluxio/uri/NoAuthority.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import com.google.common.base.Objects;
+
+/**
+ * A no authority implementation.
+ */
+public final class NoAuthority implements Authority {
+  private static final long serialVersionUID = -208199254267143112L;
+
+  public static final NoAuthority INSTANCE = new NoAuthority();
+
+  private NoAuthority() {} // to prevent initialization
```

Comment:
nit: we usually comment `// prevent instantiation` when we write private constructors for utils classes that should never be instantiated. But `NoAuthority` does get instantiated on line 22 and users are expected to use the instance. A comment like `// enforce singleton pattern` would be more accurate